src/pip/_internal/commands/install.py

```diff
@@ -267,6 +267,32 @@ def run(self, options: Values, args: List[str]) -> int:
         if options.use_user_site and options.target_dir is not None:
             raise CommandError("Can not combine '--user' and '--target'")
 
+        # Prevent users from running pip install pip install, since this likely
+        # indicates a mistake with the command
+        if args[:2] == ["pip", "install"]:
+            raise CommandError(
+                "\n".join(
+                    [
+                        "Likely incorrect command: pip install pip install ...",
+                        f"Did you mean \"pip install {' '.join(args[2:])}\"?",
+                        'To install the "install" package, run '
+                        '"pip install install" separately',
+                    ]
+                )
```

Comment:
The `join` call is unnecessary since you can use implicit string join instead. Also I think this only blocks `pip install pip install` specifically, so instead of recommending installing `install` separately, it may be better to recommend moving `pip` to another command (which is a generally good advice even without this PR anyway).